analyses/snv-callers/util/plot_functions.R

```diff
@@ -0,0 +1,274 @@
+# Plotting functions for MAF data
+#
+# C. Savonen for ALSF - CCDL
+# 2019
+#
+################################################################################
+############################# Plotting Functions ###############################
+################################################################################
+
+base_change_plot <- function(vaf_df, exp_strategy = "BOTH", filter_cutoff = 0) {
+  # Plot the number of base changes as a barplot. Need a MAF data.frame that
+  # that has `vaf`, `base_change` and `experimental_strategy` columns that are
+  # added with the `set_up_maf` function.
+  #
+  # Args:
+  #   vaf_df: MAF formatted data that has been turned into a data.frame and has
+  #           been run through `set_up_maf`.
+  #   exp_strategy: argument to specify whether to plot `wgs`, `wxs` samples or
+  #                 `both` Case insensitive.
+  #   filter_cutoff: A numeric number to only keep groups larger than it.
+  #
+  # Returns:
+  # A barplot with the number of mutations with each type of base change noted
+  # in the `base_change` column made in `set_up_maf`
+  # Make this argument case insensitive
+  exp_strategy <- toupper(exp_strategy)
+
+  # Filter out based on experimental stategy if specified
+  if (exp_strategy != "BOTH") {
+    vaf_df <- vaf_df %>%
+      dplyr::filter(experimental_strategy == exp_strategy)
+  }
+
+  # Count the number of each type of base change
+  base_count <- vaf_df %>%
```

Comment:
I don't like data frames having the same names as columns within, as it gets quite confusing (especially as in the call to ggplot below).

Reply:
Ah. Good point.